superset/utils.py

```diff
@@ -841,3 +841,17 @@ def ensure_path_exists(path):
     except OSError as exc:
         if not (os.path.isdir(path) and exc.errno == errno.EEXIST):
             raise
+
+
+def get_limit_from_sql(sql):
+    sql = sql.lower()
+    limit = None
+    tokens = sql.split()
+    try:
+        if 'limit' in tokens:
+            limit_pos = tokens.index('limit') + 1
```

Comment:
Oh that looks brittle here, for example `limit` can exist within a subquery, or there can be line breaks or tabs as separators instead of spaces. We should use the same method / regex approach as the one I defined. Maybe that method in db_engine_spec can receive `max_limit` and only apply alter if the max
_limit is lower

Reply:
Ah, true. I hadn't thought of the subquery case. But split can handle the tabs and newlines just fine. I'll go ahead to reuse your regex approach as that is more robust. 
And your suggestion of adding the logic within the checking function makes sense.